yt_dlp/extractor/rinsefm.py

```diff
@@ -71,19 +86,37 @@ class RinseFMArtistPlaylistIE(RinseFMBaseIE):
         'playlist_mincount': 7,
     }]
 
-    def _entries(self, data):
-        for episode in traverse_obj(data, (
-            'props', 'pageProps', 'episodes', lambda _, v: determine_ext(v['fileUrl']) in MEDIA_EXTENSIONS.audio),
-        ):
-            yield self._parse_entry(episode)
+    def _entries(self, episodes):
+        for episode in episodes or []:
+            # Filter out episodes without valid file URLs
+            file_url = episode.get('fileUrl')
+            if file_url and determine_ext(file_url) in MEDIA_EXTENSIONS.audio:
+                yield self._parse_entry(episode)
 
     def _real_extract(self, url):
         playlist_id = self._match_id(url)
-        webpage = self._download_webpage(url, playlist_id)
-        title = self._og_search_title(webpage) or self._html_search_meta('title', webpage)
-        description = self._og_search_description(webpage) or self._html_search_meta(
-            'description', webpage)
-        data = self._search_nextjs_data(webpage, playlist_id)
+
+        api_url = f'{self._API_BASE}/shows/{playlist_id}'
+        self.write_debug(f'API URL: {api_url}')
+
+        api_data = self._download_json(
+            api_url, playlist_id,
+            note='Downloading show data from API',
+        )
+
+        self.write_debug(f'API response keys: {list(api_data.keys()) if api_data else None}')
+
+        show_entry = traverse_obj(api_data, ('entry',))
+        episodes = traverse_obj(api_data, ('episodes',))
+
+        self.write_debug(f'Found {len(episodes) if episodes else 0} episodes')
+
+        if not episodes:
+            raise ExtractorError('Could not extract episodes from API response')
+
+        title = traverse_obj(show_entry, ('title', {str}))
+        description = traverse_obj(show_entry, ('extract', {str}))
```

Comment:
Again, I don't see the point to this excessive logging
```suggestion
        api_data = self._download_json(
            f'{self._API_BASE}/shows/{playlist_id}', playlist_id,
            note='Downloading show data from API')
```